src/compiler/checker.ts

```diff
@@ -23872,10 +23872,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isMemberOfStringMapping(source: Type, target: Type): boolean {
-        if (target.flags & (TypeFlags.String | TypeFlags.Any)) {
+        if (target.flags & TypeFlags.Any) {
             return true;
         }
-        if (target.flags & TypeFlags.TemplateLiteral) {
+        if (target.flags & (TypeFlags.String & TypeFlags.TemplateLiteral)) {

```

Comment:
Surely you meant `TypeFlags.String | TypeFlags.TemplateLiteral`.

Reply:
Apparently we don't have a test for this case.

Reply:
Shoot, yes. Probably why the test fails.

Reply:
Bad copy paste, I had it the right way before!